### PR TITLE
Define an user agent in the Github Action

### PR DESCRIPTION
**Pull request by gdelgadot**
Signed-off-by: Gloria Delgado <gloriad@vmware.com>

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -31,7 +31,10 @@ export const vibClient = clients.newClient({
       : constants.DEFAULT_VIB_PUBLIC_URL
   }`,
   timeout: 10000,
-  headers: { "Content-Type": "application/json", "User-Agent": "VIB/0.1" },
+  headers: {
+    "Content-Type": "application/json",
+    "User-Agent": "vib-action/0.1-dev",
```
Comment (recena) [inline src/main.ts:36]: We should have a parameter for the version. If not, when we cut a new release, we could forget to update this string.

Comment (gdelgadot) [reply src/main.ts:36]: Then something like this? :

Comment (gdelgadot) [reply src/main.ts:36]: Something like this? 

`"User-Agent": "vib-action"
"Version":"0.1-dev"`

Comment (mpermar) [reply src/main.ts:36]: Nah. What @recena  is suggesting is to remove the 0.1-dev altogether and pick that information dynamically. 

I've made a quick test in the old repository creating a release ( 0.1.0-dev tag ) and then printing the env vars and it seems that the version number of the GitHub Action will come as an environment variable:

Screen Shot 2022-02-22 at 11 39 39

So @gdelgadot essentially you need to change that 0.1-dev with whatever comes in the environment variable or some default (e.g. "unknown") if nothing is found.

File: src/main.ts [hunk 2]
```diff
@@ -37,7 +37,7 @@ export const vibClient = clients.newClient({
   timeout: 10000,
   headers: {
     "Content-Type": "application/json",
-    "User-Agent": `vib-action/${process.env.GITHUB_ACTION_REF}`,
+    "User-Agent": `vib-action/${userAgentVersion}`,
```
Comment (recena) [inline src/main.ts:40]: Do you mean `version`? This is not the user-agent version, but the version of the GitHub Action.

Comment (mpermar) [reply src/main.ts:40]: This change makes sense to me. But, do you want the same version for all releases?

**Review by mpermar — APPROVED** [review]

**Comment by mpermar** [comment]
Need to fix these conflicts.

File: src/main.ts [hunk 3]
```diff
@@ -14,6 +14,10 @@ const root =
     ? path.join(process.env.GITHUB_WORKSPACE, ".") // Running on GH but not tests
     : path.join(__dirname, "..") // default, but should never trigger
 
+const userAgentVersion = process.env.GITHUB_ACTION_REF
+  ? process.env.GITHUB_ACTION_REF
```
Comment (recena) [inline src/main.ts:18]: What is the value of `env.GITHUB_ACTION_REF` when the action is running?

File: src/main.ts (hunk 2 above)
Comment (recena) [reply src/main.ts:40]: I don't. We need to identify the GH version through the user-agent header.

Comment (gdelgadot) [reply src/main.ts:40]: ok, so in this case it should be ok changing ``vib-action/${userAgentVersion}`` with `via-action/${ghActionVersion}` and also the constant, right?

**Comment by gdelgadot** [comment]
I'm having issues resolving conflicts, so I will close this PR and will create a new one

**Pull request closed without merging**